### PR TITLE
Adding support to the test server to receive carved files

### PR DESCRIPTION
**Pull request by jedi22**
For the upcoming file carving functionality of osquery, the test server needed to be updated to handle receiving files from the client.

### CODE REVIEW

File: tools/tests/test_http_server.py [hunk 1]
```diff
@@ -190,6 +205,33 @@ def test_read_requests(self):
...
+    def continue_carve(self, request):
+        # Do we already have this block
+        if request['block_id'] in FILE_CARVE_MAP[request['session_id']]['blocks_received']:
+            return
+        # Store block data
+        FILE_CARVE_MAP[request['session_id']]['blocks_received'][int(request['block_id'])] = request['data']
+        # Do we still need more blocks
+        if len(FILE_CARVE_MAP[request['session_id']]['blocks_received']) < FILE_CARVE_MAP[request['session_id']]['block_count']:
+            return
+        f = open(FILE_CARVE_DIR+FILE_CARVE_MAP[request['session_id']]['carve_guid']+'.tar', 'w')
```
Comment (poppyseedplehzr) [inline tools/tests/test_http_server.py]: You might want to open this with `'wb'` As we're receiving binary data?

**Comment by jedi22** [comment]
`| 1488917975 | e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855 | 499123712 | /Users/obelisk/Downloads/archlinux-2017.03.01-dual.iso       | SUCCESS | 438c9918-12e9-46c2-82a0-8684f17eb82a | 0     |`

Test server can handle large(ish) files too

**Comment by osqueryer** [comment]
:-1: The commit d774b862142a9f0b5621097f87f074e4215d9c5b (Job results: [4222](https://jenkins.osquery.io/job/osqueryPullRequestLinux/4222/)) failed one or more tests (**Linux**).

**Comment by osqueryer** [comment]
:-1: The commit d774b862142a9f0b5621097f87f074e4215d9c5b (Job results: [4223](https://jenkins.osquery.io/job/osqueryPullRequestLinux/4223/)) failed one or more tests (**Linux**).

**Pull request merged**